MIGRATION_GUIDE.md

```diff
@@ -0,0 +1,54 @@
+# Migration Guide
+Changes are grouped as follows:
+- `Dependency` for new hard requirements that you must make sure work with your project before upgrading
+- `Deprecated` for methods/features that are still available, but will be removed in a future release
+- `Removed` for methods/features that have been removed (and what to do now!)
+- `Function Signature` for methods/functions with a change to its function signature, e.g. change in default values
+- `Functionality` for changes that are not backwards compatible
+- `Changed` for changes that do not fall into any other category
+- `Optional` for new, optional methods/features that you should be aware of - *and could take advantage of*
+
+## From v4 to v5
+
+### Dependency
+- Required dependency, `protobuf`.
```

Comment:
Should this mention optional dependency `numpy`?

Reply:
Agreed, added note in `2ab8a0f`